libs/designer/src/lib/core/state/tokens/tokensSlice.ts

```diff
@@ -55,24 +55,9 @@ export const tokensSlice = createSlice({
       // delete state.outputTokens[id]; // TODO: This causes lots of errors as tokens are not null-safe
       delete state.variables[id];
     },
-    updateVariableInfo: (state, action: PayloadAction<{ id: string; name?: string; type?: string }>) => {
-      const { id, name, type } = action.payload;
-      const variables = getRecordEntry(state.variables, id);
-      if (!variables) {
-        return;
-      }
-      if (name) {
-        state.variables[id] = variables.map((variable) => ({
-          ...variable,
-          name,
-        }));
-      }
-      if (type) {
-        state.variables[id] = variables.map((variable) => ({
-          ...variable,
-          type,
-        }));
-      }
+    updateVariableInfo: (state, action: PayloadAction<{ id: string; variables: VariableDeclaration[] }>) => {
+      const { id, variables } = action.payload;
+      state.variables[id] = variables;
```

Comment:
Can we use `getRecordEntry()` here again

Reply:
So the difference before was that we were updating the variables to point to the right name/type for downstream nodes whenever the corresponding parameter changes; however, now because the payload is not what changed but the variables themselves, we don't actually access any variables in the state here, but rather just replace them altogether

Reply:
This may be something we might want to investigate separately, as there is something that may be a concern, because the parameter is a single Editor now, when creating a new variable and changing the name, the type validation also happens; that being said, I haven't thought of a great way of doing this just yet as validation happens on the parameter level, but we'd need to bring this down to an input level